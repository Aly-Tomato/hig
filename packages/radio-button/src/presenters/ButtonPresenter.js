/* eslint react/forbid-prop-types: 0 */

import React, { Component } from "react";
import { ThemeContext } from "@hig/theme-context";
import { cx, css } from "emotion";

import stylesheet from "./stylesheet";

export default class ButtonPresenter extends Component {
  render() {
    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => {
          const styles = stylesheet(this.props, resolvedRoles);
          return (
            <span
              className={cx(
                "radio-button__wrapper",
                css(styles.radioButtonWrapper)
              )}
            />
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
