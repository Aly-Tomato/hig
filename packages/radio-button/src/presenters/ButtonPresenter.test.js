import React from "react";
import renderer from "react-test-renderer";
import ButtonPresenter from "./ButtonPresenter";

describe("radio-button/presenters/ButtonPresenter", () => {
  const resolvedRoles = {
    "checkbox.borderWidth": "3px"
  };

  [
    {
      description: "renders without props",
      props: {
        resolvedRoles
      }
    },
    {
      description: "renders with all props",
      props: {
        resolvedRoles,
        checked: true,
        disabled: true
      }
    }
  ].forEach(({ description, props: { children, ...otherProps } }) => {
    it(description, () => {
      const presenter = (
        <ButtonPresenter {...otherProps}>{children}</ButtonPresenter>
      );
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
