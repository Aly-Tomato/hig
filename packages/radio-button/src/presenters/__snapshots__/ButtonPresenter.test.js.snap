// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio-button/presenters/ButtonPresenter renders with all props 1`] = `
.emotion-0 {
  opacity: 0.5;
  position: relative;
  color: transparent;
  border: solid 1px #BEC8D2;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
}

<span
  className="radio-button__wrapper emotion-0"
/>
`;

exports[`radio-button/presenters/ButtonPresenter renders without props 1`] = `
.emotion-0 {
  opacity: 1.0;
  position: relative;
  color: transparent;
  border: solid 1px #BEC8D2;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
}

<span
  className="radio-button__wrapper emotion-0"
/>
`;
